mozilla/multi-account-containers
# Container sync

## Steps to test:

Test for first install or update of new MAC with sync:
1. Pull and load as temporary extension (on a new profile probably?)
2. Add one container to your default four
3. in extension console, run `browser.resetMAC1()`
   This adds a few configs to your `browser.storage.local` to setup with some default MAC configs
4. reload extension
   This shows the first sync upon install/update
5. Info backed up in sync is listed in console.
6. Reload again to see what would happen if data is the same as synced data in pre-synced MAC

Next test for syncing with 2nd version of FF/MAC:
1. in extension console run `browser.resetMac2()`
2. If you want, see that some icons have changed color and some containers have changed assignments.
3. reload extension to see what will happen when 2nd browser with MAC previously installed and configured is synced.
4. See that icons have changed back and new site assignments have been added.
5. Rejoice (hopefully)
6. Give lots of feedback on how to make this better

Test for syncing based on UUID after browser has already been synced once
1. add a new container or uncomment the line in resetMAC3 that does it for you
2. run `browser.resetMAC3()` then reload the extension to see the behavoir

## Notes:

 * Doesn't pass tests due to linter (console.logs), and it passes the other tests, ~~but in ugly ways, so that needs to be fixed.~~
 * `resetMac1()` and `resetMac1()` should be made into mocha tests, probably.
 * If there is a mismatch in container that a site is assigned to in addon vs. sync, the user will need to be notified to make a choice. That is not implemented yet.
 * ~~There is at least a 3rd path that still needs to be written: When MAC has been synced before. This should be more straight forward using the new macUUID to keep everything straight.~~
 * ~~container/site assignment deletion has not yet been tacked.~~
* the initial syncs (from 2 pre-existing MAC addons) are still a little wonky--you usually end up with double of everything.


Thanks!
